map/src/main/java/openlr/map/utils/GeometryUtils.java

```diff
@@ -316,50 +316,23 @@ private static double hMult(final double y) {
     public static double calculateLineBearing(final Line line,
                                               final BearingDirection dir, final int pointDistance,
                                               final int projectionAlongLine) {
-        if (line == null) {
+        if (line == null || projectionAlongLine < 0 || projectionAlongLine > line.getLineLength()) {
             return -1.0;
         }
         GeoCoordinates p1 = null;
         GeoCoordinates p2 = null;
 
+        BearingDestinationRouting bearingDestinationRouting = BearingDestinationRouting.withConfig(line, pointDistance, projectionAlongLine);
+
         if (dir == BearingDirection.IN_DIRECTION) {
-            if (projectionAlongLine > 0) {
-                p1 = line.getGeoCoordinateAlongLine(projectionAlongLine);
-                if (line.getLineLength() < projectionAlongLine + pointDistance) {
-                    p2 = line.getEndNode().getGeoCoordinates();
-                } else {
-                    p2 = line
-                            .getGeoCoordinateAlongLine(projectionAlongLine
-                                    + pointDistance);
-                }
-            } else {
-                p1 = line.getStartNode().getGeoCoordinates();
-                if (line.getLineLength() < pointDistance) {
-                    p2 = line.getEndNode().getGeoCoordinates();
-                } else {
-                    p2 = line.getGeoCoordinateAlongLine(pointDistance);
-                }
-            }
+            p1 = line.getGeoCoordinateAlongLine(projectionAlongLine);

```

Comment:
You could move this line out of the `if` block since it is repeated in the `else` block.

Reply:
482ec5f58091993383b0ec513ceafb5223fa03d7